src/orange/orangeinp/ConvexRegion.hh

```diff
@@ -216,10 +216,48 @@ class GenTrap final : public ConvexRegionInterface
     using VecReal2 = std::vector<Real2>;
     //!@}
 
+    struct TrapFace
+    {
+        real_type hy_{};  // vertical distance between horizontal edges
+        real_type hx_lo_{};  // top horizontal edge
+        real_type hx_hi_{};  // botton horizontal edge
+
+        // tan(alpha), where alpha is the clockwise angle between the
+        // _centers_ of horizontal edges, with respect to the vertical
+        // (alpha=0)
+        real_type tan_alpha_{};
+
+        TrapFace(real_type hy,
+                 real_type hx_lo,
+                 real_type hx_hi,
+                 real_type tan_alpha)
+            : hy_{hy}, hx_lo_{hx_lo}, hx_hi_{hx_hi}, tan_alpha_{tan_alpha}
+        {
+            CELER_VALIDATE(hx_lo > 0,
+                           << ": nonpositive lower x half-edge: " << hx_lo);
+            CELER_VALIDATE(hx_hi > 0,
+                           << ": nonpositive lower x half-edge: " << hx_hi);
+            CELER_VALIDATE(hy > 0,
+                           << ": nonpositive y half-projection: " << hy);
+        }
```

Comment:
Instead of validating this in the constructor (which doesn't catch errors if you modify the public data after the fact!), either declare an `explicit operator bool()` (and then `CELER_VALIDATE(lo, ...);` in `GenTrap::from_trap`; or define a `void validate() const;` or (maybe best?) simply put the `CELER_VALIDATE` in the `GenTrap::from_trap` (looping over `for (TrapFace const* tf : {&lo, &hi})`)